l10n_fr_ecotaxe_sale/tests/test_sale_ecotaxe.py

```diff
@@ -0,0 +1,124 @@
+# © 2021 Akretion (http://www.akretion.com)
+#   @author Mourad EL HADJ MIMOUNE <mourad.elhadj.mimoune@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo.addons.account.tests.common import AccountTestInvoicingCommon
+
+
+class TestsaleEcotaxe(AccountTestInvoicingCommon):
+    @classmethod
+    def setUpClass(cls, chart_template_ref="l10n_fr.l10n_fr_pcg_chart_template"):
+        super(TestsaleEcotaxe, cls).setUpClass(chart_template_ref)
+
+        cls.ecotaxe_classification = cls.env["account.ecotaxe.classification"]
+        cls.ecotaxe_classification1 = cls.ecotaxe_classification.create(
+            {
+                "name": "Fixed Ecotaxe",
+                "ecotaxe_type": "fixed",
+                "default_fixed_ecotaxe": 2.4,
+                "ecotaxe_product_status": "M",
+                "ecotaxe_supplier_status": "FAB",
+            }
+        )
+        cls.ecotaxe_classification2 = cls.ecotaxe_classification.create(
+            {
+                "name": "Weight based",
+                "ecotaxe_type": "weight_based",
+                "ecotaxe_coef": 0.04,
+                "ecotaxe_product_status": "P",
+                "ecotaxe_supplier_status": "FAB",
+            }
+        )
+        cls.product_a.weight = 100
+        cls.product_a.ecotaxe_classification_id = cls.ecotaxe_classification1.id
+        cls.product_b.weight = 400
+        cls.product_b.ecotaxe_classification_id = cls.ecotaxe_classification2.id
+
+    def create_sale_partner(self, partner_id, product_id, sale_amount=0.00):
```

Comment:
i think sale_amount isn't needed to set price_unit on so line...but without that getting totally different vals on assertion . Can't find how this method looked like historically because it was removed.